sdk/core/azure-core-management/src/main/java/com/azure/core/management/provider/IdentifierProvider.java

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.core.management.provider;
+
+import java.time.OffsetDateTime;
+
+/**
+ * The IdentifierProvider to help generate distinct values.
+ */
+public interface IdentifierProvider {
+    /**
+     * Get a random name.
+     *
+     * @param prefix the prefix to be used if possible
+     * @param maxLen the max length for the random generated name
+     * @return the random name
+     */
+    String randomName(String prefix, int maxLen);
+
+    /**
+     * Create a random UUID.
+     *
+     * @return the UUID string.
+     */
+    String randomUuid();
+
+    /**
+     * @return the current time offset.
+     */
+    OffsetDateTime dateTimeNow();
```

Comment:
This method appears only used in sample or test. So it appears not needed in core-mgmt. (worst case, disable the playback in sample, or set a date and let user change it when they run the sample).